### PR TITLE
[CS-4932]: Show dapp name and transaction chain on wc request 2.0

### PR DESCRIPTION
**Pull request by danibonilha**
### Description
This PR shows the chain and it's native token balance on the transactionSheet, for that, we needed a new instance of web3 
to independently check the balances (Let's observe the app behavior if lots of disconnected happens or pollingTracker errors), then if it's troublesome we can revisit and improve the new instance creation, there are some refactors on a useAccountProfile,  and I've switched the etherWeb3Provider setting order, so it doesn't lock the user on gnosis when switching network fails, although I think we can add a better handling for that but tbd.

### Screenshots

https://user-images.githubusercontent.com/20520102/205321157-f884c724-3608-4854-92c8-4adc7ea7b0cf.mov

### CODE REVIEW

File: cardstack/src/components/NetworkBadge/NetworkBadge.tsx [hunk 1]
```diff
@@ -8,19 +8,21 @@ import { useRainbowSelector } from '@rainbow-me/redux/hooks';
...
+    getConstantByNetwork('name', props.network || currentNetwork)
   ).toUpperCase();
 
   return (
     <Container flexDirection="row" {...props}>
       <Container
         backgroundColor="backgroundLightGray"
         paddingHorizontal={2}
-        style={{ paddingVertical: 1 }}
+        paddingVertical={1}
```
Comment (oidoug) [inline cardstack/src/components/NetworkBadge/NetworkBadge.tsx:25]: Just to point it out, we were using style prop so we could pad just 1 pixel vertically, now the banner has a 4px padding. If you think it looks better this way, I don't mind the change.

File: cardstack/src/components/TransactionConfirmationSheet/displays/GenericDisplay.tsx [hunk 2]
```diff
@@ -2,30 +2,53 @@ import { getConstantByNetwork } from '@cardstack/cardpay-sdk';
...
-};
+export const GenericDisplay = ({
+  txNetwork,
+  messageRequest,
+}: TransactionConfirmationDisplayProps) => (
+  <>
+    <FromSection txNetwork={txNetwork} />
+    <HorizontalDivider />
+    <Container width="100%">
+      <SectionHeaderText>MESSAGE</SectionHeaderText>
```
Comment (oidoug) [inline cardstack/src/components/TransactionConfirmationSheet/displays/GenericDisplay.tsx:23]: supernit: extract string

File: src/handlers/web3.js [hunk 3]
```diff
@@ -54,13 +54,13 @@ export const etherWeb3SetHttpProvider = async network => {
...
 export const getEtherWeb3Provider = async (network = undefined) => {
-  let wsConnected = web3Provider.provider?.connected;
+  let wsConnected = web3Provider?.provider?.connected;
 
   // check websocket state and reconnect if disconnected
   while (!wsConnected) {
     const currentNetwork = network || (await getNetwork());
     await etherWeb3SetHttpProvider(currentNetwork);
-    wsConnected = web3Provider.provider?.connected;
+    wsConnected = web3Provider?.provider?.connected;
```
Comment (oidoug) [inline src/handlers/web3.js:63]: I wonder if web3Provider is still undefined at this point, shouldn't we try to (re)instantiate it?

**Review by oidoug — APPROVED** [review]
LGTM 🚀  
Really nice improvement, a step towards simultaneous network support.
Tested, all looking good.

File: cardstack/src/components/TransactionConfirmationSheet/displays/GenericDisplay.tsx (hunk 2 above)
Comment (danibonilha) [reply cardstack/src/components/TransactionConfirmationSheet/displays/GenericDisplay.tsx:23]: There are lot's of MESSAGE/FROM strings across the project we should do it on a diff task

File: src/handlers/web3.js (hunk 3 above)
Comment (danibonilha) [reply src/handlers/web3.js:63]: This file is .js, I've just added a couple of optional chaining to make sure nothing breaks, there is already a loop to check if it's connected and it tries to (re)instantiate

**Pull request merged**